hook_test.go

```diff
@@ -108,6 +120,47 @@ func TestRunHookTimeoutFailure(t *testing.T) {
 	}
 }
 
+func TestRunHookWaitFailure(t *testing.T) {
+	hook := createHook(60)
+
+	hook.Args = append(hook.Args, "1", "panic")
+
+	err := hook.runHook()
+	if err == nil {
+		t.Fatal()
+	}
+}
+
+func testRunHookInvalidCommand(t *testing.T, timeout int) {
+	cleanUp()
+
+	dir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.RemoveAll(dir)
```

Comment:
This won't be called because if we exit on a t.Fatal(), this will cause to exit directly, without going through the defer stack.